sound/soc/codecs/rt700-sdw.c

```diff
@@ -490,6 +492,34 @@ static int __maybe_unused rt700_dev_suspend(struct device *dev)
 	return 0;
 }
 
+static int __maybe_unused rt700_dev_system_suspend(struct device *dev)
+{
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);
+	struct rt700_priv *rt700 = dev_get_drvdata(dev);
+	int ret;
+
+	if (!rt700->hw_init)
+		return 0;
+
+	/*
+	 * prevent new interrupts from being handled after the
+	 * deferred work completes and before the parent disables
+	 * interrupts on the link
+	 */
+	mutex_lock(&rt700->disable_irq_lock);
```

Comment:
@plbossart  if rt700_interrupt_callback gets disable_irq_lock and rt700_dev_system_suspend is waiting for this lock, jack_detect_work will be scheduled. Do we need to cancel_delayed_work_sync(&rt700->jack_detect_work) like rt700_dev_suspend does ?

Reply:
You're right @RanderWang , that's what we do implicitly by calling rt700_dev_suspend at line 520.
This last jack detection is not useful and might delay the suspend if the workqueue already started (cancel_work_sync either prevents the work function from starting, or waits until is finished) but conceptually there is nothing broken.